### PR TITLE
162 move the repository module of the mediator coordination crate to a separate crate

### PR DESCRIPTION
**Pull request by ndefokou**
@Hermann-Core @IngridPuppet Please review this PR

### CODE REVIEW

**Comment by Hermann-Core** [comment]
Please ensure you are up to date with the **main** branch before requesting again. There’re conflicts.

**Review by Hermann-Core — CHANGES_REQUESTED** [review]

File: crates/plugins/mediator-coordination/src/web/handler.rs [hunk 1]
```diff
@@ -102,7 +104,7 @@ pub mod tests {
         let repository = AppStateRepository {
             connection_repository: Arc::new(MockConnectionRepository::from(vec![])),
             secret_repository: Arc::new(MockSecretsRepository::from(vec![])),
-            message_repository: Arc::new(MockMessagesRepository::from(vec![])),
+            message_repository: Arc::new(MockMessagesRepository::from(vec![])), // Merged from origin
```
[the remark this answers is not in the document]
Comment (Hermann-Core) [reply crates/plugins/mediator-coordination/src/web/handler.rs:107]: @ndefokou check at the manifest file and remove all the unused dependencies. 
https://github.com/adorsys/didcomm-mediator-rs/blob/35bc4f2185db4638a4f0c6d9ee67275e23d89a14/crates/database/Cargo.toml#L6-L25
The **`[feature]`** and **`[dev-dependencies]`** sections are not needed. Delete them.
https://github.com/adorsys/didcomm-mediator-rs/blob/35bc4f2185db4638a4f0c6d9ee67275e23d89a14/crates/database/Cargo.toml#L27-L37

File: crates/database/src/lib.rs [hunk 2]
```diff
@@ -5,7 +5,7 @@ use thiserror::Error;
 use mongodb::error::Error as MongoError;
 /// A trait representing an abstract resource.
 /// Any type implementing this trait should also implement `Serialize`.
-pub trait Entity: Sized + Serialize {}
+pub trait Entity: Sized + Serialize {}
```
[the remark this answers is not in the document]
Comment (IngridPuppet) [reply crates/database/src/lib.rs]: Hey @Hermann-Core - It is a shared abstraction for all the objects to store in MongoDB collections. It helps constrain a set of operations to implement for associated repositories.

[the remark this answers is not in the document]
Comment (IngridPuppet) [reply crates/database/src/lib.rs]: Fine then.

File: crates/did-utils/examples/ed25519-sign.rs [hunk 3]
```diff
@@ -1,6 +1,7 @@
 extern crate did_utils;
-use did_utils::crypto::{Ed25519KeyPair, {Generate, CoreSign}};
-
+use did_utils::crypto::{
+    Ed25519KeyPair, {CoreSign, Generate},
```
[the remark this answers is not in the document]
Comment (ndefokou) [reply crates/did-utils/examples/ed25519-sign.rs:3]: I am not working with this file i did a `cargo fmt` on the entire project

**Review by Hermann-Core — APPROVED** [review]

**Pull request merged**